Non-JS header menu & search

janehughes opened the pull request:
<!-- ignore-task-list-start -->

### What is the context of this PR?

Ticket: https://jira.ons.gov.uk/browse/ONSDESYS-434

Progressively enhances the header menu and search element by setting them to display by default, then hiding them with JS once the JS is loaded. This also updates the menu styles slightly to match the designs (see ticket) when js is disabled, and also updated the tests to ensure they pass (and removes a redundant test to check for aria attribute on a hidden element @ desktop).

### How to review this PR

- Review [header with search button](http://localhost:3001/components/header/example-header-with-search-button.html) variant locally 
**Note:** The scope of progressive enhancement in this ticket is _only_ for this menu & search header variant, so the other header variants/navs should not be affected.
- Disable JS on your browser & see that the menu & search display as expected (refer to designs)
**Note:** I've updated the menu and search toggle to also use the active styling, to show menus are open for users as they would when JS is enabled - this isn't reflected in the designs.
- Enable JS and see that the menu & search collapse automatically on load and then expand/hide as expected when interacting with them as usual
- Review the rest of the header variants and ensure they display & function as expected across both mobile & desktop views and remain unaffected

### Checklist

This needs to be completed by the person raising the PR.

<!-- ignore-task-list-end -->

-   [x] I have selected the correct Assignee
-   [x] I have linked the correct Issue

rmccar commented:
When you turn js off, the design of these buttons doesn't seem to match the design in the screenshot on the ticket. They are also still focusable via the keyboard are they meant to be focusable? I feel like if they cant be clicked then they shouldn't be?

<img width="250" alt="Screenshot 2025-05-01 at 12 02 02" src="https://github.com/user-attachments/assets/578b7f42-ec25-4dda-a466-2680af274c98" />

admilne reviewed (commented):
LGTM. I was about to approve when I saw the comments from Rich. I think he raises a valid point over whether or not the menu and search buttons should be focusable if you can't toggle them on or off.

janehughes commented:
> When you turn js off, the design of these buttons doesn't seem to match the design in the screenshot on the ticket. They are also still focusable via the keyboard are they meant to be focusable? I feel like if they cant be clicked then they shouldn't be?
> 
> <img alt="Screenshot 2025-05-01 at 12 02 02" width="250" src="https://private-user-images.githubusercontent.com/42928680/439580923-578b7f42-ec25-4dda-a466-2680af274c98.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDYxOTg2NzEsIm5iZiI6MTc0NjE5ODM3MSwicGF0aCI6Ii80MjkyODY4MC80Mzk1ODA5MjMtNTc4YjdmNDItZWMyNS00ZGRhLWE0NjYtMjY4MGFmMjc0Yzk4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTAyVDE1MDYxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFjZDE3MTIxNGNhOTlhYjJlYTkxNzgxMWY5NmQ2MDJlMzY5NDk1N2IzMWJlODRlOTNlNTFhZDIwNDI5ZWFlMzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.hskiG9dpL2DPrGnMApSlUPBvn_ba3NOdlVJXVNOEj0A">

Really good point, I've updated this but rather than removing the focus state I've adjusted the active styling with a 'disabled' class and also added the `aria-disabled` attribute to signify the buttons are not functional when JS isn't loaded. I've kept the focus styles in, as having done a bit of research it seems important to always keep focus state visible as it's helpful for users to keep track of their place on the page and also understand the context/functionality (especially for keyboard/SR users), even if the buttons are currently inactive.

cc @admilne too

janehughes commented on src/components/navigation/navigation.js:
```diff
@@ -92,15 +93,21 @@ export default class NavigationToggle {
+            // JS is active - always hide by default
+            this.closeNav();
+        } else {
+            // JS inactive
```
Updated now, thanks @MaciekBaron :)

MaciekBaron reviewed (approved):
LGTM!

rmccar reviewed (changes requested):
This seems fine now but I think we need to add some tests around the new active and disabled classes and aria-disabled attribute

janehughes commented:
> This seems fine now but I think we need to add some tests around the new active and disabled classes and aria-disabled attribute

Thanks @rmccar I've added several new tests to check the new atttributes both before & after JS is enabled

rmccar commented:
I think the new JS tests are better off in the script test file (header.spec.js). Also through this Ive just noticed that a lot of the js changes for the menu and search buttons have been added to the navigation component rather than the header component. I think this might need a bit of a refactor to add a new header JS file and move the JS for this into there. It doesn't make sense for it to be in the navigation component because these buttons aren't using that. But the tests themselves look good

janehughes commented:
> I think the new JS tests are better off in the script test file (header.spec.js). Also through this Ive just noticed that a lot of the js changes for the menu and search buttons have been added to the navigation component rather than the header component. I think this might need a bit of a refactor to add a new header JS file and move the JS for this into there. It doesn't make sense for it to be in the navigation component because these buttons aren't using that. But the tests themselves look good

For the sake of progressing this on the website's side, would it be okay if we merge this as-is and I split this refactoring into a new tidy up/enhancement ticket? As I think this might take a while to do and test and feels big enough to make it a separate piece of work.

rmccar commented:
> > I think the new JS tests are better off in the script test file (header.spec.js). Also through this Ive just noticed that a lot of the js changes for the menu and search buttons have been added to the navigation component rather than the header component. I think this might need a bit of a refactor to add a new header JS file and move the JS for this into there. It doesn't make sense for it to be in the navigation component because these buttons aren't using that. But the tests themselves look good
> 
> For the sake of progressing this on the website's side, would it be okay if we merge this as-is and I split this refactoring into a new tidy up/enhancement ticket? As I think this might take a while to do and test and feels big enough to make it a separate piece of work.

Yeah ok thats fine
